py-server/test_default_model.py

```diff
@@ -0,0 +1,38 @@
+import json
+import unittest
+from main import build_message_history
+from default_model import DefaultModel
+from vertexai.language_models import ChatMessage
+from langchain_core.messages import HumanMessage, AIMessage
+
+
+# Create a test class for the build_message_history function
+class TestDefaultModel(unittest.TestCase):
+
+    def setUp(self):
+        self.model = DefaultModel()
+
+    def test_build_message_history(self):
+        incoming_messages = json.dumps(
+            [
+                {"content": "Message 1", "author": "user"},
+                {"content": "Message 2", "author": "llm"},
+                {"content": "Message 3", "author": "user"},
+                {"content": "Message 4", "author": "llm"},
+            ]
+        )
+
+        expected_messages = [
+            ChatMessage("Message 1", "user"),
+            ChatMessage("Message 2", "llm"),
+            ChatMessage("Message 3", "user"),
+            ChatMessage("Message 4", "llm"),
+        ]
+
+        message_history = build_message_history(incoming_messages)
```

Comment:
we want to test `DefaultModel.convert_message_history`, but currently this is also testing `build_message_history`.  I suggest we remove the call to `build_message_history` (which is already unit tested) and explicitly test the type conversion. 


```py
incoming_messages = [
        HumanMessage("Message 1"),
        AIMessage("Message 2"),
        HumanMessage("Message 3"),
        AIMessage("Message 4"),
     ]
vertex_message_history = self.model.convert_message_history(message_history)
self.assertEqual(vertex_message_history, expected_messages)

```

Reply:
done